cluster-install/install.sh

```diff
@@ -2532,11 +2542,34 @@ if trident_will_be_installed_or_modified; then
         if components_include_trident; then
             # Trident upgrade (includes operator upgrade if needed)
             if trident_image_needs_upgraded; then
-                if config_trident_image_is_custom || prompt_user_yes_no "Would you like to upgrade Trident?"; then
-                    step_generate_torc_patch "$_EXISTING_TORC_NAME" "$(get_config_trident_image)" "" "" "$(get_config_trident_autosupport_image)"
-                    trident_operator_image_needs_upgraded && step_generate_trident_operator_patch
-                else
-                    loginfo "Trident will not be upgraded."
+                # if trident version < 23.10
+                if version_higher_or_equal "23.07" "$_EXISTING_TRIDENT_VERSION"; then
+                    logwarn "Your Trident installation is at version 23.07, while the lowest required version to enable ACP is 23.10."
+                fi
+
+                if config_trident_image_is_custom; then
+                    local -r custom_image="$(as_full_image "$TRIDENT_IMAGE_REGISTRY" "$TRIDENT_IMAGE_REPO" "$TRIDENT_IMAGE_TAG")"
+                    local warning_message="Warning: we cannot verify the version of the custom Trident image you provided"
+                    warning_message += " ($custom_image). ACP support cannot be guaranteed if upgrading to that image."
+                    logwarn "$warning_message"
+                fi
+
+                if ! components_include_connector && (components_include_trident || components_include_acp); then
+                    if prompt_user_yes_no "Would you like to upgrade Trident? Choosing no will exit the script (yes/no):"; then
+                        step_generate_torc_patch "$_EXISTING_TORC_NAME" "$(get_config_trident_image)" "" "" "$(get_config_trident_autosupport_image)"
+                        trident_operator_image_needs_upgraded && step_generate_trident_operator_patch
+                    else
+                        exit 0
+                    fi
+                fi
+
+                if components_include_connector; then
```

Comment:
Should probably check to see if ACP is already enabled here. It's possible the user has a Trident upgrade available yet already has ACP enabled.

Reply:
Added this below